client/state/states-list/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	STATES_LIST_RECEIVE,
+	STATES_LIST_REQUEST,
+	STATES_LIST_REQUEST_FAILURE,
+} from 'state/action-types';
+import { statesListSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+// Stores the complete list of states, indexed by locale key
+export const items = createReducer( {}, {
+	[ STATES_LIST_RECEIVE ]: ( state, action ) => ( { ...state, [ action.countryCode ]: action.statesList } ),
+}, statesListSchema );
+
+// Tracks states list fetching state
+export const isFetching = createReducer( false, {
```

Comment:
Your first argument (default state) should be an object now with changes in 4bbf4c3.